internal/restic/node_windows.go

```diff
@@ -70,26 +70,109 @@ func (node Node) restoreSymlinkTimestamps(path string, utimes [2]syscall.Timespe
+		extr.Name = attr.Name
+		extr.Value = attr.Value
+		eas = append(eas, *extr)
+	}
+	if len(eas) > 0 {
+		if errExt := restoreExtendedAttributes(node.Type, path, eas); errExt != nil {
+			return errExt
+		}
+	}
+	return nil
 }
 
-// Listxattr retrieves a list of names of extended attributes associated with the
-// given path in the file system.
-func Listxattr(path string) ([]string, error) {
-	return nil, nil
-}
+// fill extended attributes in the node. This also includes the Generic attributes for windows.
+func (node *Node) fillExtendedAttributes(path string, _ bool) (err error) {
+	var fileHandle windows.Handle
 
-func IsListxattrPermissionError(_ error) bool {
-	return false
-}
+	//Get file handle for file or dir
+	if node.Type == "file" {
+		if strings.HasSuffix(filepath.Clean(path), `\`) {
+			return nil
+		}
+		utf16Path := windows.StringToUTF16Ptr(path)
+		fileAccessRightReadWriteEA := (0x8 | 0x10)
+		fileHandle, err = windows.CreateFile(utf16Path, uint32(fileAccessRightReadWriteEA), 0, nil, windows.OPEN_EXISTING, windows.FILE_ATTRIBUTE_NORMAL, 0)
+	} else if node.Type == "dir" {
+		utf16Path := windows.StringToUTF16Ptr(path)
+		fileAccessRightReadWriteEA := (0x8 | 0x10)
+		fileHandle, err = windows.CreateFile(utf16Path, uint32(fileAccessRightReadWriteEA), 0, nil, windows.OPEN_EXISTING, windows.FILE_ATTRIBUTE_NORMAL|windows.FILE_FLAG_BACKUP_SEMANTICS, 0)
+	} else {
+		return nil
+	}
+	if err != nil {
+		err = errors.Errorf("open file failed for path: %s, with: %v", path, err)
+		return err
+	}
+	defer func() {
+		err := windows.CloseHandle(fileHandle)
+		if err != nil {
+			debug.Log("Error closing file handle for %s: %v\n", path, err)
+		}
+	}()
+
+	//Get the windows Extended Attributes using the file handle
+	extAtts, err := fs.GetFileEA(fileHandle)
+	debug.Log("fillExtendedAttributes(%v) %v", path, extAtts)
+	if err != nil {
+		debug.Log("open file failed for path: %s : %v", path, err)
+		return err
+	} else if len(extAtts) == 0 {
+		return nil
+	}
+
+	//Fill the ExtendedAttributes in the node using the name/value pairs in the windows EA
+	for _, attr := range extAtts {
+		if err != nil {
+			err = errors.Errorf("can not obtain extended attribute for path %v, attr: %v, err: %v\n,", path, attr, err)
+			continue
+		}
+		extendedAttr := ExtendedAttribute{
+			Name:  attr.Name,
+			Value: attr.Value,
+		}
 
-// Setxattr associates name and data together as an attribute of path.
-func Setxattr(path, name string, data []byte) error {
+		node.ExtendedAttributes = append(node.ExtendedAttributes, extendedAttr)
+	}
 	return nil
 }
 
+// restoreExtendedAttributes handles restore of the Windows Extended Attributes to the specified path.
+// The Windows API requires setting of all the Extended Attributes in one call.
+func restoreExtendedAttributes(nodeType, path string, eas []fs.ExtendedAttribute) (err error) {
+	var fileHandle windows.Handle
+	switch nodeType {
+	case "file":
+		utf16Path := windows.StringToUTF16Ptr(path)
```

Comment:
Why does `fillExtendedAttributes` check for `if strings.HasSuffix(filepath.Clean(path), '\') {`, but not this function? Please extract the helper code to open a file handle, it seems to be identical between `fillExtendedAttributes` and `restoreExtendedAttributes`.

Reply:
Removing strings.HasSuffix(filepath.Clean(path), `\`) check, it is not needed here.
Created common helper function.